keylime/keylime_agent.py

```diff
@@ -488,6 +489,14 @@ def main(argv=sys.argv):
         logger.critical("This process must be run as root.")
         return
 
+    if config.get('cloud_agent', 'agent_uuid') == 'dmidecode':
+        ret = cmd_exec.run('which dmidecode', raiseOnError=False)
+        if ret['code'] != 0 or os.getuid() != 0:
```

Comment:
Is it possible to split this condition out? Check first that we are root and give an error message for that, then check that dmidecode is available and error on that. Seems better to have 2 different error codes to the 2 different conditions rather than combine them together.

Reply:
sure, doable